video/out/hwdec/hwdec_vaapi.c

```diff
@@ -359,6 +362,43 @@ static bool try_format_map(struct ra_hwdec *hw, struct mp_image *surface)
     return ok;
 }
 
+static bool try_format_upload(struct ra_hwdec *hw, int mp_fmt)
+{
+    bool success = false;
+    if (IMGFMT_IS_HWACCEL(mp_fmt))
+        return success;
+
+    struct mp_image *src = mp_image_alloc(mp_fmt, 2, 2);
+    if (!src)
+        return success;
+
+    struct priv_owner *p = hw->priv;
+    AVBufferRef *hw_pool = av_hwframe_ctx_alloc(p->ctx->av_device_ref);
+
+    // Arbitrarily use YUV420P for the hw_subfmt. Attempt a hwupload to seee
+    // if vaExportSurfaceHandle works.
+    mp_update_av_hw_frames_pool(&hw_pool, p->ctx->av_device_ref, IMGFMT_VAAPI,
+                                IMGFMT_420P, src->w, src->h, false);
+
```

Comment:
I'm not convinced that this is sufficient. Firstly, a driver could simply not support YUV420P surfaces (or not support them with composed layers), and then it might not support uploading a source format that might be uploadable to some other destination format. Your first iteration of this code which tested all pairwise combinations is the most thorough way of doing this.

Reply:
Yeah as much as I would love to test all pairwise combinations, I nuked that implementation because it was horrendously slow. Delayed startup time by nearly a second.

I don't know if it's any consolation but when going through the debugging output for all possible pairwise combinations, I found that in *all* scenarios the upload either failed or succeeded for all possible combinations. The only thing that mattered was the format of the source image. If it could be used, all conversions to all other hw_subfmts worked. If not, they all failed.

Reply:
Does setting `export AMD_DEBUG=noefc` make a difference?

Reply:
The slowness was on my laptop which is intel. I think it's just because most formats work there and then you end up testing every single possible combination by doing a hwupload which is just slow. Thankfully, I haven't seen any evidence testing thus far that we need to do that.

Reply:
Ok. If it looks like there is no practical difference, then I'm ok with this.